src/input4mips_validation/cli/__init__.py

```diff
@@ -1,7 +1,48 @@
 """
 Command-line interface
 """
-# Have to import everything in here so that the root_cli registers
-# that it has other commands.
-from input4mips_validation.cli.root import root_cli  # noqa: F401
-from input4mips_validation.cli.validate_file import validate_file_command  # noqa: F401
+from pathlib import Path
+from typing import Annotated
+
+import typer
+
+from input4mips_validation.cli.cli_logging import setup_logging
+from input4mips_validation.validation import assert_file_is_valid
+
+app = typer.Typer()
+
+
+@app.callback()
+def main() -> None:
+    """
+    Entrypoint for the command-line interface
+    """
+    setup_logging()
+
+
+@app.command(name="validate-file")
+def validate_file_command(
+    filepath: Annotated[
+        Path, typer.Argument(exists=True, dir_okay=False, file_okay=True)
+    ],
```

Comment:
Cool, I didn't know about `Annotated`. Definitely clearer than the old way of writing
```
filepath: Path = typer.Argument(…)
```
which looked like `filepath` had a default value.

Reply:
Nice, thanks!

Reply:
Nice, thanks!